crates/nu-system/src/foreground.rs

```diff
@@ -0,0 +1,53 @@
+// It's a simpler version for fish shell's external process handling.
+//
+// For more information, please check `child_setup_process` function in fish shell.
+// https://github.com/fish-shell/fish-shell/blob/3f90efca38079922b4b21707001d7bb9630107eb/src/postfork.cpp#L140
+#[cfg(target_family = "unix")]
+pub mod external_process_setup {
+    use std::os::unix::prelude::CommandExt;
+    pub fn prepare_to_foreground(external_command: &mut std::process::Command) {
+        unsafe {
+            libc::signal(libc::SIGTTOU, libc::SIG_IGN);
+            libc::signal(libc::SIGTTIN, libc::SIG_IGN);
+
+            external_command.pre_exec(|| {
+                // make the command startup with new process group.
+                // The process group id must be the same as external commands' pid.
+                // Or else we'll failed to set it as foreground process.
+                // For more information, refer to `fork_child_for_process` function:
+                // https://github.com/fish-shell/fish-shell/blob/023042098396aa450d2c4ea1eb9341312de23126/src/exec.cpp#L398
+                libc::setpgid(0, 0);
+                Ok(())
```

Comment:
Can we add SAFETY comment for `unsafe`

Reply:
Oh, that's nice :-D   We definitely need comments for `unsafe` block.